android-record/src/main/java/com/fasterxml/jackson/module/androidrecord/AndroidRecordModule.java

```diff
@@ -0,0 +1,167 @@
+      /**
+       * Names of actual Record components from definition; auto-detected.
+       */
+      private final Set<String> _componentNames;
+
+      private AndroidRecordNaming(MapperConfig<?> config, AnnotatedClass forClass) {
+          super(config, forClass,
+                  // no setters for (immutable) Records:
+                  null,
+                  // trickier: regular fields are ok (handled differently), but should
+                  // we also allow getter discovery? For now let's do so
+                  "get", "is", null);
+          _componentNames = getDesugaredRecordComponents(forClass.getRawType()).map(Field::getName)
+                  .collect(Collectors.toSet());
+      }
+
+      @Override
+      public String findNameForRegularGetter(AnnotatedMethod am, String name)
+      {
+          // By default, field names are un-prefixed, but verify so that we will not
+          // include "toString()" or additional custom methods (unless latter are
+          // annotated for inclusion)
+          if (_componentNames.contains(name)) {
+              return name;
+          }
+          // but also allow auto-detecting additional getters, if any?
+          return super.findNameForRegularGetter(am, name);
+      }
+  }
+
+  private static class AndroidRecordClassIntrospector extends BasicClassIntrospector {
+    @Override
+    protected POJOPropertiesCollector collectProperties(MapperConfig<?> config, JavaType type, MixInResolver r,
+                                                        boolean forSerialization) {
+      if (isDesugaredRecordClass(type.getRawClass())) {
+        AnnotatedClass classDef = _resolveAnnotatedClass(config, type, r);
+        AccessorNamingStrategy accNaming = new AndroidRecordNaming(config, classDef);
+        return constructPropertyCollector(config, classDef, type, forSerialization, accNaming);
+      }
+
+      return super.collectProperties(config, type, r, forSerialization);
+    }
+  }
+
+  @Override
+  public void setupModule(SetupContext context) {
+    super.setupModule(context);
+    context.addValueInstantiators(AndroidRecordModule::findValueInstantiator);
+    context.setClassIntrospector(new AndroidRecordClassIntrospector());
+  }
+
+  static boolean isDesugaredRecordClass(Class<?> raw) {
+    return raw.getSuperclass() != null && raw.getSuperclass().getName().equals("com.android.tools.r8.RecordTag");
+  }
+
+  private static ValueInstantiator findValueInstantiator(DeserializationConfig config, BeanDescription beanDesc,
+                                                         ValueInstantiator defaultInstantiator) {
+    Class<?> raw = beanDesc.getType().getRawClass();
+    if (! defaultInstantiator.canCreateFromObjectWith() && defaultInstantiator instanceof StdValueInstantiator
+            && isDesugaredRecordClass(raw)) {
+      Map<String, Type> components = getDesugaredRecordComponents(raw)
+              .collect(Collectors.toMap(Field::getName, Field::getGenericType));
```

Comment:
Instead of `getGenericType()`, just use (type-erased) `getType()` -- matching is done using type-erased types.

Reply:
Hmmh. How is this possible? All matching in class files typically uses erased types, and cannot really use type parameters to resolve conflicts.

Unless, I guess, this is combined with parameter ordering differing. That's probably it and makes sense. So since Constructor parameter ordering not fixed wrt fields, can have same type erased set in different order.